demo/api/src/db/fixtures/generators/blocks/dam-image.fixture.ts

```diff
@@ -0,0 +1,54 @@
+import { ExtractBlockInputFactoryProps } from "@comet/blocks-api";
+import { DamImageBlock, FocalPoint, ImageCropAreaInput } from "@comet/cms-api";
+import { Injectable } from "@nestjs/common";
+import { datatype, random } from "faker";
+
+import { PixelImageBlockFixtureService } from "./pixel-image.fixture";
+import { SvgImageBlockFixtureService } from "./svg-image.fixture";
+
+@Injectable()
+export class DamImageBlockFixtureService {
+    constructor(
+        private readonly svgImageBlockFixture: SvgImageBlockFixtureService,
+        private readonly pixelImageBlockFixture: PixelImageBlockFixtureService,
+    ) {}
+
+    async generateBlock(): Promise<ExtractBlockInputFactoryProps<typeof DamImageBlock>> {
+        let type: "svgImage" | "pixelImage" = "pixelImage";
+        let props = await this.pixelImageBlockFixture.generateBlock();
+
+        if (datatype.boolean()) {
+            type = "svgImage";
+            props = await this.svgImageBlockFixture.generateBlock();
+        }
```

Comment:
I think it would be more understandable if you'd define `attachedBlocks` and make a random array pick. However, this would create all blocks all the time (even now pixel image block is needlessly created  when it's an svg image block)

Reply:
I changed it to:
```
let type: "svgImage" | "pixelImage";
let props;

if (datatype.boolean()) {
    type = "svgImage";
    props = await this.svgImageBlockFixtureService.generateBlock();
} else {
    type = "pixelImage";
    props = await this.pixelImageBlockFixtureService.generateBlock();
}

return {
    attachedBlocks: [
        {
            type,
            props,
        },
    ],
    activeType: type,
};
```

or would you prefer it like this?
```
type AttachedDamImageBlocks = Array<{
    type: "svgImage" | "pixelImage";
    props: ExtractBlockInputFactoryProps<typeof SvgImageBlock | typeof PixelImageBlock>;
}>;

// Class code ...
async generateBlock(): Promise<ExtractBlockInputFactoryProps<typeof DamImageBlock>> {
    const attachedBlocks: AttachedDamImageBlocks = [
        { type: "svgImage", props: await this.svgImageBlockFixtureService.generateBlock() },
        { type: "pixelImage", props: await this.pixelImageBlockFixtureService.generateBlock() },
    ];

    const activeBlock = faker.random.arrayElement(attachedBlocks);

    return {
        attachedBlocks: attachedBlocks,
        activeType: activeBlock.type,
    };
}
```

Reply:
What about the following third option?

```ts
const type = faker.random.arrayElement(["svgImage", "pixelImage"]);
let props;

switch (type) {
  case "svgImage":
    props = await this.svgImageBlockFixtureService.generateBlock();
    break;
  case "pixelImage":
    props = await this.pixelImageBlockFixtureService.generateBlock();
    break;
}

return {
    attachedBlocks: [
        {
            type,
            props,
        },
    ],
    activeType: type,
};
```

Would be more easy to extend, and removes the `if (datatype.boolean())`, which I don't like

Reply:
hmm typescript throws an error with this code.

typescript infers the type of props like this:
<img width="702" alt="Screenshot 2024-01-02 at 16 04 37" src="https://github.com/vivid-planet/comet/assets/44967610/9a6fd6ac-f874-4ced-8ff7-739a2778d449">

and then throws this error
<img width="808" alt="Screenshot 2024-01-02 at 16 06 20" src="https://github.com/vivid-planet/comet/assets/44967610/67c6b26a-e105-4df2-a799-9a768768b462">

this could be an alternative:
<img width="945" alt="Screenshot 2024-01-04 at 08 42 09" src="https://github.com/vivid-planet/comet/assets/44967610/55fc8726-955e-4aaf-80a3-d9a35c01a649">

or saving it into the attached blocks as well:
<img width="952" alt="Screenshot 2024-01-02 at 16 08 13" src="https://github.com/vivid-planet/comet/assets/44967610/e4bb9387-478f-468b-a33d-d6924fc963e2">

Reply:
I'm fine with either alternative